src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotSource.java

```diff
@@ -419,7 +431,8 @@ private Result<?, UpdateError> handleScheduledTrip(
       .createUpdatedTripTimes(tripUpdate, timeZone, serviceDate, backwardsDelayPropagationType);
 
     if (result.isFailure()) {
-      return result;
+      // needs to be put into a new Result so the success type is correct
+      return Result.failure(result.failureValue());
```

Comment:
Could we expand Result with this kind of method, returning a new Result with the correct types directly, or possibly just returning the same instance?

Reply:
Done in 124ced049b11d1d17cdd6481e0a60a204f90cbd2